enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/LogViewMgr.java

```diff
@@ -1114,11 +1121,9 @@ public void stateChanged(final FetchLogEvent event) {
             switch (event.getState()) {
                 case COMPLETED: case FAILED:
                     FetchLog oldLog = null;
-                    synchronized (serverInputStreams) {
-                        FetchLog storedLog = serverInputStreams.get(instance);
-                        if (this.log.equals(storedLog)) {
-                            oldLog = serverInputStreams.remove(instance);
-                        }
+                    FetchLog storedLog = serverInputStreams.computeIfPresent(instance, (key, value) -> value);
+                    if (this.log.equals(storedLog)) {
+                        oldLog = serverInputStreams.remove(instance);
```

Comment:
Changes semantics(race condition). Before operations in lines 1118-1121 were atomic, after a different thread might change `serverInputStreams` and/or `oldLog` while the lines are executed.

Reply:
You are correct.